### PR TITLE
fix tiled symbol rendering for viewport-collision

### PR DESCRIPTION
**Pull request by ansis**
This replaces `MapMode::Still` with `MapMode::Static` and `MapMode::Tile`. `Tile` makes sure the symbols in the resulting tile are tileable while symbols in `Still` match rendering in `Continuous` mode.

-js pr that adds two render tests: https://github.com/mapbox/mapbox-gl-js/pull/5650

### CODE REVIEW

File: src/mbgl/renderer/renderer_impl.cpp [hunk 1]
```diff
@@ -358,7 +358,7 @@ void Renderer::Impl::render(const UpdateParameters& updateParameters) {
                 // We only need clipping when we're _not_ drawing a symbol layer. The only exception
                 // for symbol layers is when we're rendering still images. See render_symbol_layer.cpp
                 // for the exception we make there.
-                if (!symbolLayer || parameters.mapMode == MapMode::Still) {
+                if (!symbolLayer || parameters.mapMode != MapMode::Tile) {
```
Comment (ChrisLoer) [inline src/mbgl/renderer/renderer_impl.cpp]: We don't clip symbol layers at all anymore, right? So should this just be `if (!symbolLayer)`, and the comment should be updated to remove the reference to the still image exception?

File: platform/node/src/node_map.cpp [hunk 2]
```diff
@@ -1074,6 +1074,15 @@ NodeMap::NodeMap(v8::Local<v8::Object> options)
                            ->NumberValue()
                      : 1.0;
       }())
+    , mode([&] {
```
Comment (jfirebaugh) [inline platform/node/src/node_map.cpp]: Maybe a `"mode" option with possible values `"static"` and `"tile"`? Also, can you add a note about this change to the node CHANGELOG? api-gl will need an update.

Comment (ansis) [reply platform/node/src/node_map.cpp]: switched in https://github.com/mapbox/mapbox-gl-native/pull/10443/commits/b93ed14090fde76d3d4c13d2fd472bca44745ed1

**Comment by ChrisLoer** [comment]
It looks like you already pushed this to `viewport-collision`

**Pull request closed without merging**